test/TestBuffer.sol

```diff
@@ -74,4 +75,126 @@ contract TestBuffer {
     uint64 actualBig = buf.readUint64();
     Assert.equal(uint(actualBig), uint(big), "Read multiple data chunks from the same Buffer (inner cursor works as expected)");
   }
+
+  function testReadUint8asUint16() external {
+    ThrowProxy throwProxy = new ThrowProxy(address(this));
+    bool r;
+    bytes memory error;
+    uint8 expected = 0xAA;
```

Comment:
We are actually expecting this to fail, so I guess the values defined here are more `input` or something similar rather than `expected`.